website/src/plugins/github-service.ts

```diff
@@ -0,0 +1,113 @@
+import { Octokit } from '@octokit/rest';
+
+import type { GitHubMetadata } from './github-metadata';
+
+export class GitHubService {
+  private octokit: Octokit;
+  private owner: string;
+  private repo: string;
+
+  constructor(owner: string, repo: string) {
+    this.octokit = new Octokit();
```

Comment:
suggestion: Maybe we should be able to pass the Octokit as a parameter, so we can jsut mock it without having it to mock the real request ? or have it as an optional constructor parameter so tests can specify it

Reply:
I was following Florent's comment here: https://github.com/podman-desktop/podman-desktop/pull/13182#pullrequestreview-3022059612 but I agree that both solutions work

Reply:
cc @benoitf IMO we should use provide the octokit client through the constructor, it would avoid adding `msw` as a dependency, and only work with mocks instead of having to mock networks call, and would simplify the tests

Reply:
it's always the tricky question of mocks and usage

Here it's an easy option to mock the data and still use the library that gets the data (Octokit)

vs

you mock everything

At the end, you can have unit tests that pass but it doesn't work at all, so I like somehow to validate the real calls made by Octokit